### PR TITLE
Initial pytests starting with TSP trace client; pylint issue fix for dictionary appends

### PR DESCRIPTION
**Pull request by marco-miller**

### CODE REVIEW

**Review by MatthewKhouzam — APPROVED** [review]

**Comment by MatthewKhouzam** [comment]
These patches look good, but the pytest patch patch is a bit non-trivial, I would like Bernd's once over.

**Comment by marco-miller** [comment]
> These patches look good, but the pytest patch patch is a bit non-trivial, I would like Bernd's once over.

Ack; thanks. Will soon push follow-up commit(s) as well.

**Review by MatthewKhouzam — APPROVED** [review]
I like this a lot, I don't know if it's possible to put a warning at the first 404 going: Hey! Turn on the server! But it works well. Let's wait for Bernd on this one?

File: test_tsp.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        response = self.tsp_client.fetch_traces()
+        assert len(response.model.traces) == 2
+        # Delete each previously opened trace to clean server for next run.
+        for trace in response.model.traces:
+            # Do not also delete the trace from disk; file part of this repo.
+            response = self.tsp_client.delete_trace(trace.UUID, False)
+            assert response.status_code == 200
+
+    def test_fetch_opened_trace_then_delete(self, traces_path):
+        response = self.tsp_client.open_trace('channel0_0', traces_path)
```
Comment (MatthewKhouzam) [inline test_tsp.py:66]: One thing, it's kinda the convention to open the 'metadata' file. They all work though

Comment (marco-miller) [reply test_tsp.py:74]: > One thing, it's kinda the convention to open the 'metadata' file. They all work though

Thanks. I need to look deeper into this, as I'm not familiar with this subtlety yet.

**Comment by marco-miller** [comment]
> I like this a lot, I don't know if it's possible to put a warning at the first 404 going: Hey! Turn on the server! But it works well. Let's wait for Bernd on this one?

In the meantime, I just pushed the follow-up commit f56af7e herein, to try addressing this good concern; thanks.

File: test_tsp.py (hunk 1 above)
Comment (marco-miller) [reply test_tsp.py]: -Done. I amended a few commits to add a second test trace path fixture, also fixing the opened trace names accordingly. Now each test trace path (namely kernel's and kernel_vm's) should properly be opened. Both paths or file contents are now treated as an actual trace each.

Thanks again and here is the diff for review: https://github.com/theia-ide/tsp-python-client/compare/f56af7e7fbd3e17b7c21f29a40da78a169f06ef5..170c551ef0662a361e6a9886e28333647e7ce964

**Comment by bhufmann** [comment]
@marco-miller I had to manually fetch the test traces submodule ("git submodule update --init --recursive"), and with that the tests execution worked fine. Does everybody need to do that to run the test or do I miss something?

**Comment by MatthewKhouzam** [comment]
I didn't need --recursive... but I did that init.

**Comment by marco-miller** [comment]
> @marco-miller I had to manually fetch the test traces submodule ("git submodule update --init --recursive"), and with that the tests execution worked fine. Does everybody need to do that to run the test or do I miss something?

Good catch; and thanks for the review. Done through efbc524.

**Comment by marco-miller** [comment]
> I didn't need --recursive... but I did that init.

Indeed; thanks.

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks for the contribution.

**Pull request merged**